### PR TITLE
Implement filter: brightness.

### LINKED ISSUE #421: WR hits unreachable code on MotionMark
By running Servo on http://browserbench.org/MotionMark/ with `window.matchMedia = function() {return {}}` in `resources/user-agent-js/00.example.js`, we hit [this `unreachable!()`](https://github.com/servo/webrender/blob/3e17aa58710feabc3e7d77b0d792db8beaf9fa92/webrender/src/tiling.rs#L1988)

```
$ ./mach run -r http://browserbench.org/MotionMark/ --userscripts -w
internal error: entered unreachable code (thread <unnamed>, at /Users/manishearth/mozilla/servo/./.cargo/git/checkouts/webrender-c3596abe1cf4f320/master/webrender/src/tiling.rs:1988)
stack backtrace:
   0:        0x10ae94c3e - backtrace::backtrace::trace::h0e60ef08c7c34e9f
   1:        0x10ae94f2c - backtrace::capture::Backtrace::new::h8bf319c36d8f5d1b
   2:        0x109a631f4 - servo::main::_{{closure}}::h2d2b5bc7b4eac464
   3:        0x10b331943 - std::panicking::rust_panic_with_hook::hb1322e5f2588b4db
   4:        0x10aeea114 - std::panicking::begin_panic::h0914615a412ba184
   5:        0x10af0c40a - webrender::frame::Frame::flatten::h51e44fdb58c5ae23
   6:        0x10af0c02d - webrender::frame::Frame::flatten::h51e44fdb58c5ae23
   7:        0x10af0c02d - webrender::frame::Frame::flatten::h51e44fdb58c5ae23
   8:        0x10af206f5 - webrender::render_backend::RenderBackend::build_scene::h52afb899b760e4ed
   9:        0x10af17751 - webrender::render_backend::RenderBackend::run::h02e7b47b1bbd522c
  10:        0x10aeea7ef - std::panicking::try::do_call::ha940a6c78ff65cb4
  11:        0x10b332c4a - __rust_maybe_catch_panic
  12:        0x10aef43f6 - _<F as alloc..boxed..FnBox<A>>::call_box::h7a759c70df46c07e
  13:        0x10b330864 - std::sys::thread::Thread::new::thread_start::h022e3887023c6290
  14:     0x7fff8ee9299c - _pthread_body
  15:     0x7fff8ee92919 - _pthread_start
ERROR:servo: internal error: entered unreachable code
```

### PR DESCRIPTION
**Pull request by glennw**
Fixes #421.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/webrender/422)

### CODE REVIEW

**Comment by glennw** [comment]
r? @pcwalton 

cc @Manishearth @jdm

**Comment by pcwalton** [comment]
I'm a bit confused: why isn't this handled like the other filters?

**Comment by glennw** [comment]
The other filters currently run through the composite path, which is a (performance) bug. The composite path has both source textures available, but filters only need one. Long term, I need to convert the rest of the filters to go through this same path, and rename ps_blend to ps_filter.

File: webrender/src/tiling.rs [hunk 1]
```diff
@@ -2190,9 +2196,15 @@ impl Default for PackedStackingContext {
 }
 
 #[derive(Debug, Copy, Clone)]
+enum SimpleCompositeInfo {
+    Opacity(f32),
+    Brightness(f32),
+}
+
+#[derive(Debug, Copy, Clone)]
 enum CompositeKind {
```
Comment (pcwalton) [inline webrender/src/tiling.rs]: Add a small comment here explaining what the difference is between simple and complex.

**Pull request merged**